### PR TITLE
[enhancement](Nereids)support count, min and avg function

### PR DESCRIPTION
**Pull request by yinzhijian**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:
1. add count function
2. add min function
3. add avg function
## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
4. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by jackwener — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSlotReference.java [hunk 1]
```diff
@@ -282,6 +282,10 @@ public BoundStar(List<Slot> children) {
             );
         }
 
+        public String toSql() {
+            return children.stream().map(Expression::toSql).collect(Collectors.joining(","));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSlotReference.java:286]: nit:
```suggestion
            return children.stream().map(Expression::toSql).collect(Collectors.joining(", "));
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Avg.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Override
+    public Expression withChildren(List<Expression> children) {
+        Preconditions.checkArgument(children.size() == 1);
+        return new Avg(children.get(0));
+    }
+
+    @Override
+    public DataType getIntermediateType() {
+        return getDataType();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Avg.java:54]: avg's IntermediateType is not consistent with Function data type

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindFunction.java [hunk 3]
```diff
@@ -90,6 +93,24 @@ public Expression visitUnboundFunction(UnboundFunction unboundFunction, Void con
                     return unboundFunction;
                 }
                 return new Sum(unboundFunction.getArguments().get(0));
+            } else if (name.equalsIgnoreCase("count")) {
+                List<Expression> arguments = unboundFunction.getArguments();
+                if (arguments.size() != 1) {
+                    return unboundFunction;
+                }
+                return new Count(unboundFunction.getArguments().get(0));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindFunction.java:101]: could we handle count(*) and count(1) now?

**Comment by 924060929** [comment]
good job, you can add some regression test suite like this?

```groovy
suite('test_aggregate_function') {
    // use regression_test_ssb_sf1 database, you can compute ssb database by suite directory like 'regression-test/suites/ssb_sf1/nereids/q1.1.groovy'
    sql "use ${realDb}"

    sql 'set enable_nereids_planner=true'
    // nereids need vectorized
    sql 'set enable_vectorized_engine=true'

    sql 'set exec_mem_limit=2147483648*2'

    qt 'select count(*), count(l.*), count(1), count(lo_orderkey), count(l.lo_orderkey), min(lo_revenue), min(l.lo_revenue), avg(lo_revenue), avg(l.lo_revenue) from lineorder as l'
}
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Count.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.stream.Collectors;
+
+/** count agg function. */
+public class Count extends AggregateFunction {
+
+    private final boolean isStar;
+
+    public Count() {
+        super("count");
+        this.isStar = true;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Count.java:37]: we need to handle count(1) and count(*) in PruneAggChildColumns.

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**